src/jvm/main/org/jetbrains/kotlinx/lincheck/nvm/crash.kt

```diff
@@ -139,18 +142,22 @@ object Crash {
     }
 }
 
-private class BusyWaitingBarrier {
+private class BusyWaitingBarrier(private val yieldCallback: () -> Unit) {
     private val free = atomic(false)
     private val waitingCount = atomic(0)
 
     inline fun await(action: (Boolean) -> Unit) {
         waitingCount.incrementAndGet()
         // wait for all to access the barrier
-        while (waitingCount.value < Crash.threads && !free.value);
+        while (waitingCount.value < Crash.threads && !free.value) {
+            yieldCallback()
```

Comment:
Due to this callback won't you cover some odd interleavings like "switch from thread 1 to thread 2 because of a system crash, switch back to thread 1, then switch again from thread 1, because not all threads reached this barrier"?

Reply:
It is true. I've fixed it by force manual switch